src/components/icon/js/iconService.js

```diff
@@ -447,38 +447,58 @@ function MdIconService(config, $templateRequest, $q, $log, $mdUtil, $sce) {
   }
 
   /**
-   * Lookup registered fontSet style using its alias...
-   * If not found,
+   * Lookup a registered fontSet style using its alias.
+   * @param {string} alias used to lookup the alias in the array of fontSets
+   * @returns {*} matching fontSet or the defaultFontSet if that alias does not match
    */
   function findRegisteredFontSet(alias) {
     var useDefault = angular.isUndefined(alias) || !(alias && alias.length);
-    if (useDefault) return config.defaultFontSet;
+    if (useDefault) {
+      return config.defaultFontSet;
+    }
 
     var result = alias;
     angular.forEach(config.fontSets, function(it) {
-      if (it.alias == alias) result = it.fontSet || result;
+      if (it.alias === alias) {
+        result = it.fontSet || result;
+      }
     });
 
     return result;
   }
 
+  /**
+   * @param {Icon} cacheElement cached icon from the iconCache
+   * @returns {Icon} cloned Icon element with unique ids
+   */
   function transformClone(cacheElement) {
     var clone = cacheElement.clone();
     var cacheSuffix = '_cache' + $mdUtil.nextUid();
 
-    // We need to modify for each cached icon the id attributes.
-    // This is needed because SVG id's are treated as normal DOM ids
-    // and should not have a duplicated id.
-    if (clone.id) clone.id += cacheSuffix;
-    angular.forEach(clone.querySelectorAll('[id]'), function(item) {
-      item.id += cacheSuffix;
+    // For each cached icon, we need to modify the id attributes and references.
+    // This is needed because SVG ids are treated as normal DOM ids and should not be duplicated on
+    // the page.
+    if (clone.id) {
+      clone.id += cacheSuffix;
+    }
+
+    var addCacheSuffixToId = function(match, p1, p2, p3) {
+      return [p1, p2, cacheSuffix, p3].join('');
+    };
+    angular.forEach(clone.querySelectorAll('[id]'), function(descendantElem) {
+      descendantElem.id += cacheSuffix;
     });
+    // Inject the cacheSuffix into all instances of url(id) and xlink:href="#id".
```

Comment:
Can you add a comment that explains why there's no xss risk?

Reply:
Is there some way that https://github.com/angular/material/blob/master/src/core/util/util.js#L1-L7 could be vulnerable to XSS? Since that is what we'll be coercing to a `string`, sticking into `innerHTML`, and then calling `++` on.

Reply:
I pretty much meant a comment that says just that.

Guard also also still useful in cases like this, though, in case someone else down the line changes the way unique ids are generated

Reply:
OK I added a comment just below this.

I'm not clear what you meant by "Guard also also still useful". Perhaps that I should add a comment about this in `$mdUtil` as well?

Reply:
I meant something like asserting that the ID matches a certain pattern and throwing if not. Not required, but something to keep in mind.

Reply:
Good idea. Guard added!